File: api/telegram/receiver.py
import os

from telethon import TelegramClient, events

API_ID = int(os.environ["TG_API_ID"])
API_HASH = os.environ["TG_API_HASH"]
TARGET_CHAT_ID = os.environ.get("TG_TARGET_CHAT_ID")
DOWNLOAD_DIR = os.environ.get("TG_DOWNLOAD_DIR", "downloads")

if TARGET_CHAT_ID is not None:
    TARGET_CHAT_ID = int(TARGET_CHAT_ID)

os.makedirs(DOWNLOAD_DIR, exist_ok=True)

client = TelegramClient("receiver", API_ID, API_HASH)


@client.on(events.NewMessage)
async def on_new_message(event):
    if TARGET_CHAT_ID is not None and event.chat_id != TARGET_CHAT_ID:
        return

    message = event.message

    if message.message:
        print(f"[text] chat={event.chat_id}: {message.message}")

    if message.video or (message.media and getattr(message.media, "document", None)):
        saved_path = await message.download_media(file=DOWNLOAD_DIR)
        if saved_path:
            print(f"[media] saved to: {saved_path}")


async def main():
    print("Listening for Telegram messages...")
    await client.run_until_disconnected()


with client:
    client.loop.run_until_complete(main())

File: api/telegram/receiver.js
const { TelegramClient } = require("telegram");
const { NewMessage } = require("telegram/events");
const { StringSession } = require("telegram/sessions");
const input = require("input");

const apiId = parseInt(process.env.TG_API_ID || "", 10);
const apiHash = process.env.TG_API_HASH;
const stringSession = new StringSession(process.env.TG_STRING_SESSION || "");

if (!apiId || !apiHash) {
  throw new Error("Set TG_API_ID and TG_API_HASH environment variables");
}

(async () => {
  const client = new TelegramClient(stringSession, apiId, apiHash, {
    connectionRetries: 5,
  });

  await client.start({
    phoneNumber: async () => input.text("Phone number: "),
    password: async () => input.text("2FA password (if enabled): "),
    phoneCode: async () => input.text("Code: "),
    onError: (error) => console.error(error),
  });

  console.log("Logged in.");
  console.log("Session string (save for next runs):");
  console.log(client.session.save());

  client.addEventHandler((event) => {
    const msg = event.message;
    if (!msg) {
      return;
    }

    if (msg.message) {
      console.log("[text]", msg.message);
    }

    if (msg.media) {
      console.log("[media] received media");
    }
  }, new NewMessage({}));

  console.log("Listening for new messages...");
})();
